config/unix-g++.cmake

```diff
@@ -40,78 +26,41 @@ check_c_compiler_flag(   "-march=native" HAS_MARCH_NATIVE )
 
 #
 # Compiler Flags
-#
-# Consider using these diagnostic flags for Debug builds:
-# -Wcast-qual - warn about casts that remove qualifiers like const.
-# -Wstrict-overflow=4
-# -Wwrite-strings
-# -Wunreachable-code
-# - https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html
-#
+
 # Consider using these optimization flags:
 # -ffast-math -ftree-vectorize
 # -fno-finite-math-only -fno-associative-math -fsignaling-nans
 #
-# Added, but shouldn't be needed:
-# -Wno-expansion-to-defined - unable to use GCC diagnostic pragma to suppress warnings.
-#
 # Control FMA
 # -ffp-contract=off
 
 if( NOT CXX_FLAGS_INITIALIZED )
   set( CXX_FLAGS_INITIALIZED "yes" CACHE INTERNAL "using draco settings." )
-  string(APPEND CMAKE_C_FLAGS " -Wcast-align -Wpointer-arith -Wall -pedantic -Wfloat-equal"
-    " -Wunused-macros -fsanitize=bounds-strict -Wshadow -Wformat=2")
+  string(APPEND CMAKE_C_FLAGS " -Wall -Wextra -pedantic -Wcast-align -Wpointer-arith -Wfloat-equal"
+    " -Wunused-macros -Wshadow -Wformat=2")
   if( CMAKE_CXX_COMPILER_VERSION VERSION_GREATER 7.0 )
-    string( APPEND CMAKE_C_FLAGS " -Wno-expansion-to-defined -Wnarrowing -Wnull-dereference" )
+    string( APPEND CMAKE_C_FLAGS " -Wnull-dereference" )
   endif()
-  string( CONCAT CMAKE_C_FLAGS_DEBUG "-g -fno-inline -fno-eliminate-unused-debug-types -O0 -Wextra"
-    " -Wundef -Wunreachable-code -DDEBUG")
-  # Ref: https://github.com/lefticus/cppbestpractices/blob/master/02-Use_the_Tools_Available.md#compilers
-  # -Wfloat-equal       # Warn when if-statement compares floats directly
-  # -Werror             # Promote all warnings to 'error' status
-  # -Wconversion        # warn on type conversions that may lose data
-  # -Wdouble-promotion  # warn if float is implicit promoted to double
-  # -Wformat=2          # warn on security issues around functions that format output (ie printf)
-  # -Winline            # Warn if function marked 'inline' cannot be inlined
-  # -Wold-style-cast    # Warn if c-style casts are used, replace with c++-style cast.
-  set( CMAKE_C_FLAGS_RELEASE        "-O3 -funroll-loops -D_FORTIFY_SOURCE=2 -DNDEBUG" )
+  string( CONCAT CMAKE_C_FLAGS_DEBUG "-g -fno-inline -fno-eliminate-unused-debug-types -O0"
+    " -Wundef -Wunreachable-code -fsanitize=bounds-strict -fdiagnostics-color=auto -DDEBUG")
```

Comment:
Oh I see it's here in debug, does it slow down release code?

Reply:
The performance cost varies a bit. Usually ~10% slower.  That is enough to keep it out of the Optimized builds.  The generated code is also somewhat larger because barriers are created in between objects in memory.

Reply:
I didn't do a performance test in our code. That 10% number is what I see online.

Reply:
Sounds good! Thanks for explaining it.